sretoolbox/container/image.py

```diff
@@ -353,9 +382,17 @@ def _request_get(self, url):
 
             # Try again, this time with the Authorization header
             headers['Authorization'] = self._get_auth(www_auth)
-            response = requests.get(url, headers=headers)
+            response = requests.head(url, headers=headers)
+
+        if 'etag' in response.headers:
+            with contextlib.suppress(KeyError):
+                return self.response_cache[response.headers['etag']]
 
+        response = requests.get(url, headers=headers, auth=self.auth)
```

Comment:
you're passing `auth` toghether with `Authorization` headers. Not sure how safe this is.

Reply:
So far I see it working, if you have a scenario that worries you I'll make sure to test it and get it fixed.

Reply:
Yes, you are setting the `Authorization` header twice, with different data. Why are you doing that? What's the purpose of having it set twice, with different data?

Reply:
To keep the code simple and avoid branching based on whether a header has been inserted somewhere. I've updated the 401 block to discard the `auth` tuple if it isn't valid for this image.